satpy/composites/__init__.py

```diff
@@ -551,69 +551,105 @@ def _insert_palette_colors(channels, palette):
 class DayNightCompositor(GenericCompositor):
     """A compositor that blends a day data with night data."""
 
-    def __init__(self, name, lim_low=85., lim_high=88., **kwargs):
+    def __init__(self, name, lim_low=85., lim_high=88., day_night="day_night", **kwargs):
         """Collect custom configuration values.
 
         Args:
             lim_low (float): lower limit of Sun zenith angle for the
                              blending of the given channels
             lim_high (float): upper limit of Sun zenith angle for the
                              blending of the given channels
+            day_night (string): "day_night" means both day and night portions will be kept
+                                "day_only" means only day portion will be kept
+                                "night_only" means only night portion will be kept
 
         """
         self.lim_low = lim_low
         self.lim_high = lim_high
+        self.day_night = day_night
         super(DayNightCompositor, self).__init__(name, **kwargs)
 
     def __call__(self, projectables, **kwargs):
         """Generate the composite."""
         projectables = self.match_data_arrays(projectables)
 
-        day_data = projectables[0]
-        night_data = projectables[1]
+        foreground_data = projectables[0]
+        background_data = projectables[1] if self.day_night == "day_night" else foreground_data
```

Comment:
How ugly does the code if you set this to `None` if `day_night` isn't `day_night` and do various `if` statements through the rest of the code to only use it `if background_data is not None:`?

Reply:
I tried this before but there's "referenced before assignment" error so I did it this way...

Reply:
:thinking: Well that probably means you needed an extra `if` statement. Let's see what others have to say about your work before you try anything else (or I request anything else). I'll ask others for review on slack. Thanks for your work @yukaribbba.

Reply:
OK

Reply:
Now this `if` decides by length of `projectables`. Same to another one in `cos_zen`.

Reply:
Sorry I might mistake your meaning. You don't want `background_data` get involved unless `day_night` is `day_night`, right? But set it to `None` will lead to some errors during the next steps. So my solution would be `background_data = foreground_data`. I'm trying to keep the "double" structure in the current version.

Reply:
It only leads to errors if you don't have `if background_data is not None:`. Those operations should only be done if background_data is set. For example, when you do `add_bands`, if background data is None then add `'A'` to the foreground data. When combining metadata, only call `bombine_metadat` if background data is not None. You might even be able to have one giant if block like:

```python

final_metadata = foreground_data.attrs.copy()
foreground_data = add_bands(...)
if background_data is not None:
    final_metadata = combine_metadata(final_metadata, background_metadata.attrs)
    background_data = add_bands(background_data, foreground_data['bands'])

... other stuff ...
```

Again, not the biggest deal, it just avoids the extra duplicate logic of applying things to foreground_data twice.

Reply:
I understand what you said. But there's a problem in expression: I did this little test with only one composite and `"day_only"`
```
if projectables[1] is not None:
     print ("background_data YES")
else:
     print ("background_data NO")
```
I expected a "NO", while actually I got an error.
```
  File "c:\users\45107\documents\github\satpy\satpy\composites\__init__.py", line 582, in __call__
    if projectables[1] is None:
IndexError: list index out of range
```
So we want to determine whether `background_data` exists. But is it the proper way to say `None` or `not None`?

Reply:
Maybe `len(projectables) > 1` better for this?

Reply:
What about `if "only" in self.day_night:`? Or `if self.day_night != "day_night":`? `projectables` will be a list of whatever the user passed. Even if the user passes 2 or more "prerequisites" in the YAML (which is a mistake), we still only want to do the "background_data" logic when the user requested it.

Reply:
Ok, now I splitted it into two parts. Also, there seems to be no need for `match_data_arrays` since the second one is alternative.